intelmq/bots/experts/time_filter/expert.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+from datetime import datetime, timedelta
+from dateutil import parser
+from intelmq.lib.utils import get_timedelta
+from intelmq.lib.bot import Bot
+from datetime import timezone
+
+
+class TimeFilterExpertBot(Bot):
+    search_field: str = 'time.source'
+    search_from: str  = '1d'
+
+    not_after = None
+
+    def init(self):
+        self.search_field = self.search_field
+        if self.search_from:
+            timedelta_params = get_timedelta(self.search_from)
+            self.not_after = datetime.now(tz=timezone.utc) - timedelta(**timedelta_params)
+
+    def process(self):
+        event = self.receive_message()
+        # time based filtering
+        if self.search_field in event:
+            try:
+                event_time = parser.parse(str(event.get(self.search_field)))
```

Comment:
the timebased fields in the event already contain a `datetimevalue`, so you're converting those to string and back to `datetime` - is there a specific reason for this?

Reply:
Some field can be string, not date